### ISSUE TITLE
ktlint-cli baseline file should not contain absolute filenames

### ISSUE THREAD

**Issue by paul-dingemans** [pinterest/ktlint#2146]
Ktlint CLI expects the baseline file to contain errors stored per relative file. Instead the errors are store for absolute filenames. As a result Ktlint CLI does not suppress the errors for a baseline that it has created.

Reproduce:

1. Create a file `Foo.kt` containing code below:
```
fun Foo() = "foo"
```
2. Remove `baseline.xml`
3. Create new `baseline.xml` with `ktlint --baseline=baseline.xml`. Baseline contains an error for a file with an absolute path:
```
<?xml version="1.0" encoding="utf-8"?>
<baseline version="1.0">
    <file name="/some/absolute/path/project/src/main/kotlin/Foo.kt">
        <error line="1" column="5" source="standard:function-naming" />
    </file>
</baseline>
```
4. Rerun `ktlint --baseline=baseline.xml` and note that the error is reported while it should have been suppressed.
